ksa_compliance/output_models/e_invoice_output_model.py

```diff
@@ -104,6 +106,29 @@ def __init__(
             self.sales_invoice_doc.name, sales_invoice_additional_fields_doc.invoice_doctype
         )
 
+        self.branch_doc = None
+        if self.business_settings_doc.enable_branch_configuration == 1:
+            if not self.sales_invoice_doc.branch:
+                fthrow(
+                    msg=_('Branch is mandatory when ZATCA Branch configuration is enabled.'),
+                    title=_('Branch Is Mandatory'),
+                )
+
+            self.branch_doc = cast(
+                Branch,
+                frappe.get_doc(
+                    'Branch',
+                    self.sales_invoice_doc.branch,
+                ),
+            )
+            if self.branch_doc.custom_company != self.sales_invoice_doc.company:
+                fthrow(
+                    msg=_(
+                        f'Select branch {self.sales_invoice_doc.branch} is not configured for company: {self.sales_invoice_doc.company}'
```

Comment:
Translatable messages can't contain interpolated values. When switching to `ft`, you'll be able to use placeholders: `ft('Selected branch $branch is not configured for company $company', branch=self.sales_invoice_doc.branch, company=self.sales_invoice_doc.company)`